Lib/ufo2ft/outlineCompiler.py

```diff
@@ -424,7 +424,11 @@ def setupTable_OS2(self):
         os2.version = 0x0004
         # average glyph width
         widths = [width for width, _ in hmtx.metrics.values() if width > 0]
-        os2.xAvgCharWidth = round(sum(widths) / len(widths))
+        if widths:
+            os2.xAvgCharWidth = round(sum(widths) / len(widths))
+        else:
+            logger.warn("All glyphs have a zero width. Is this intentional?")
```

Comment:
Use `logger.warning(...)` instead of `logger.warn(...)` which is [deprecated](https://docs.python.org/3/library/logging.html#logging.Logger.warning). Keep the message short, remove the question "Is this intentional?".

Reply:
I think we don’t even need to warn. It’s a valid font file